integr8ly/cloud-resource-operator
# MGDAPI-6008 update aws rds certs

## Overview

Jira: MGDAPI-6008

## What
this pr checks the existing cert and updates it if it's not the new cert

## Verification
Verification is a bit difficult as all new RDS provisioned with have the latest cert 
- `make cluster/prepare`
- `PROVIDER=aws make cluster/seed/postgres`
- `make run`
- Wait for the postgres status.phase is complete
- Stop the operator ctrl-C
- confirm that the cert is `rds-ca-rsa2048-g1` in the ui or via cli
```bash
aws rds describe-db-instances --query 'DBInstances[*].CACertificateIdentifier'
[
    "rds-ca-rsa2048-g1"
]
```
- in the code here https://github.com/integr8ly/cloud-resource-operator/pull/800/files#diff-a8d4a6f2c4ee23973b06074457ca4b143e3731ac44e91850e517bc2438303a82R63 change the values around for existingCert and newCert
- modify the maintenance window in the aws to the current time in the ui or via the aws cli
```bash
# get the instance id
~ aws rds describe-db-instances --query 'DBInstances[*].DBInstanceIdentifier' --output text
<your-instance-id>
~ aws rds modify-db-instance \
    --db-instance-identifier <your-instance-id> \
    --preferred-maintenance-window Mon:00:00-Mon:03:00 \
    --apply-immediately
```
- update postgres CR spec with `maintanceWindow : true` & `applyImmediately: true` 
- Wait for the postgres status.phase is complete
- Stop the operator ctrl-C
- confirm that the cert is `rds-ca-2019` in the ui or via cli
```bash
aws rds describe-db-instances --query 'DBInstances[*].CACertificateIdentifier'
[
    "rds-ca-2019"
]
```
- This confirms that the cert will change over
